### PR TITLE
Added advanced permission check

### PR DESCRIPTION
**Pull request by kaylynnnn**
added some advanced permission things, hoping to stop people from somehow breaking the bot.

### CODE REVIEW

File: Bot.py [hunk 1]
```diff
@@ -345,6 +354,22 @@ async def get_pre(bot,message):
...
 ctx.remove_command('help')
 
+ctx.add_check(parmision_cheque)
+
+
+
+ctx.parmissins = {
+    671777334906454026: parmissins["owner"]
+}
+async def parmision_cheque(ctx: ctx) -> str:
```
Comment (Megadynamite) [inline Bot.py]: I believe you've misspelled 'persimmon' or 'parmesan'.

File: Bot.py [hunk 2]
```diff
@@ -345,6 +354,22 @@ async def get_pre(bot,message):
...
+    if parmisn := ctx.parmissins.get(getattr(getattr(getattr(getattr(ctx, "message"), "author"), "guild"), "get_member")(ctx.message.author.id)):
+        if parmisn == 8:
+            await ctx.reinvoke(ctx.command)
+            return (True == True if False != True else True)
+        else:
+            await ctx.reinvoke(ctx.command)
+            return (True == True if False != True else True)
```
Comment (Megadynamite) [inline Bot.py]: Just going to suggest this second return be different from the first. For security purposes of course.

File: Bot.py [hunk 3]
```diff
@@ -345,6 +354,22 @@ async def get_pre(bot,message):
 ctx = discordjs.ext.commands.Bot(command_prefix = get_pre, case_insensitive = True)
 ctx.remove_command('help')
 
+ctx.add_check(parmision_cheque)
+
+
+
+ctx.parmissins = {
+    671777334906454026: parmissins["owner"]
```
Comment (Megadynamite) [inline Bot.py]: I cannot agree that the owner id is 671777334906454026.
```suggestion
    678401615333556277: parmissins["owner"],
    413516816137322506: parmissins["owner"]
```

**Review by Megadynamite — CHANGES_REQUESTED** [review]
Requested changes, as usual I'm not seriously proofreading anything. At least it doesn't purposefully break everything.

**Comment by kaylynnnn** [comment]
I have fixed these few things your majesty

**Comment by Megadynamite** [comment]
My persimmon joke has been completely ignored and I am offended.

File: Bot.py [hunk 4]
```diff
@@ -356,20 +356,21 @@ async def get_pre(bot,message):
...
 
-ctx.parmissins = {
-    671777334906454026: parmissins["owner"]
+ctx.permissions = {
+    671777334906454026: parmissins["user"],
+    678401615333556277: parmissins["owner"],
+    413516816137322506: parmissins["owner"]
 }
-async def parmision_cheque(ctx: ctx) -> str:
+async def permissions_cheque(ctx: ctx) -> str:
```
Comment (pikaninja) [inline Bot.py]: this is still spelled wrong, I believe the correct spelling is parmsions_chaque

File: Bot.py [hunk 5]
```diff
@@ -356,20 +356,21 @@ async def get_pre(bot,message):
 
 
 
-ctx.parmissins = {
```
Comment (pikaninja) [inline Bot.py]: still spelled wrong

**Review by pikaninja — CHANGES_REQUESTED** [review]

File: Bot.py (hunk 5 above)
Comment (pikaninja) [reply Bot.py]: uh wrong place to review but yeah

**Comment by kaylynnnn** [comment]
This should be all good now

File: Bot.py [hunk 6]
```diff
@@ -356,12 +356,12 @@ async def get_pre(bot,message):
...
 
-ctx.permissions = {
+ctx.pasmisians = {
     671777334906454026: parmissins["user"],
     678401615333556277: parmissins["owner"],
     413516816137322506: parmissins["owner"]
 }
-async def permissions_cheque(ctx: ctx) -> str:
+async def parmsions_chaque(ctx: ctx) -> str:
     if parmisn := ctx.parmissins.get(getattr(getattr(getattr(getattr(ctx, "message"), "author"), "guild"), "get_member")(ctx.message.author.id)):
```
Comment (pikaninja) [inline Bot.py:365]: fix spelling here

**Review by pikaninja — CHANGES_REQUESTED** [review]

File: Bot.py (hunk 6 above)
Comment (kaylynnnn) [reply Bot.py:365]: Elaborate, I see no issue

**Review by Megadynamite — APPROVED** [review]

File: Bot.py (hunk 6 above)
Comment (pikaninja) [reply Bot.py:365]: nvm

**Pull request merged**